tests/ui/consts/const-mut-refs-crate.rs

```diff
@@ -0,0 +1,34 @@
+//@ run-pass
+//@ aux-build:const_mut_refs_crate.rs
+
+#![feature(const_mut_refs)]
+
+//! Regression test for https://github.com/rust-lang/rust/issues/79738
+//! Show how we are duplicationg allocations, even though static items that
+//! copy their value from another static should not also be duplicating
+//! memory behind references.
+
+extern crate const_mut_refs_crate as other;
+
+use other::{
+    inner::{INNER_MOD_BAR, INNER_MOD_FOO},
+    BAR, FOO,
+};
+
+pub static LOCAL_FOO: &'static i32 = &41;
+pub static LOCAL_BAR: &'static i32 = LOCAL_FOO;
+pub static mut COPY_OF_REMOTE_FOO: &'static mut i32 = unsafe { FOO };
+
+static DOUBLE_REF: &&i32 = &&99;
+static ONE_STEP_ABOVE: &i32 = *DOUBLE_REF;
```

Comment:
```suggestion
static mut DOUBLE_REF: &mut &mut i32 = &mut &mut 99;
static ONE_STEP_ABOVE: &mut i32 = unsafe { *DOUBLE_REF };
```
Wouldn't this be a more interesting test?

Reply:
that's what it used to be, but then I realized the same issue exists for immutable statics, and it seemed better to prevent both and test immutable references, as those are what are more commonly used

Reply:
But you're not testing the same thing, you're testing different things for both of them. I'm fine with duplicating this with and without `mut`.

However this is all in one file so I don't think it reproduces the issue with immutable statics?  This [already passes](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e744a2870feb3c1fba858caa058d417d) on stable.